rows.go

```diff
@@ -638,7 +657,13 @@ func (rs *namedStructRowScanner) appendScanTargets(dstElemValue reflect.Value, s
 				colName = sf.Name
 			}
 			fpos := fieldPosByName(fldDescs, colName)
-			if fpos == -1 || fpos >= len(scanTargets) {
+			if fpos == -1 {
```

Comment:
pretty much the only difference is adding this lax check, allowing struct field to exist without corresponding query result